src/core/Directus/Services/ItemsService.php

```diff
@@ -171,7 +171,113 @@ public function findOne($collection, array $params = [])
+            }
+        }
+        
+        $this->validatePayload($collection, null,  $collectionFields, $params);
+    }
+    
+    /**
+     * Validate Many To Many Collection Fields
+     */
+    public function validateManyToManyCollection($payload, $params, $aliasColumnDetails, $recordData){
+        $colName = $aliasColumnDetails->getName();
+        $relationalCollectionName = $aliasColumnDetails->getRelationship()->getCollectionManyToMany();
+        if($relationalCollectionName && isset($payload[$colName])){
+            $relationalCollectionPrimaryKey = SchemaService::getCollectionPrimaryKey($relationalCollectionName);
+            $relationalCollectionColumns = SchemaService::getAllCollectionFields($relationalCollectionName);
+
+            foreach($payload[$colName] as $individual){     
+                if(!isset($individual['$delete'])){ 
+                    $aliasField = $aliasColumnDetails->getRelationship()->getJunctionOtherRelatedField();
+                    $validatePayload = $individual[$aliasField];
+                    $storedData = (!empty($recordData) && isset($recordData[$colName])) ? $recordData[$colName] : [] ;
+                    
+                    foreach($relationalCollectionColumns as $column){
+                        if(!empty($recordData) && !$column->isAlias() && !$column->hasPrimaryKey() && isset($recordData[$colName])){
+                            $search = array_search($individual[$relationalCollectionPrimaryKey], array_column($storedData, $relationalCollectionPrimaryKey));
+                            $columnName = $column->getName();
+                            if($search !== false){
+                                $dbObj = isset($storedData[$search][$aliasField]) ? $storedData[$search][$aliasField] : [];
+                                $validatePayload[$columnName] = isset($validatePayload[$columnName]) ? $validatePayload[$columnName]: (isset($dbObj[$columnName]) ? $dbObj[$columnName] : null);
+                            }else{
+                                $relationalCollectionData = $this->findByIds(
+                                    $relationalCollectionName,
+                                    $validatePayload[$relationalCollectionPrimaryKey],
+                                    $params
+                                );
+                                $validatePayload[$columnName] = isset($validatePayload[$columnName]) ? $validatePayload[$columnName]: (isset($relationalCollectionData['data'][$columnName]) ? $relationalCollectionData['data'][$columnName] : null);
+                            }
+                        }
+                    }
+                    $this->validatePayload($relationalCollectionName, null, $validatePayload,$params);
+                }
+            }
+        }  
+    }
+    
+    /**
+     * Validate Alias Collection Fields
```

Comment:
As this function adds support for OneToMany, ManyToOne. Please mention in comment.